src/cookies.test.ts

```diff
@@ -0,0 +1,145 @@
+import { getCookies, getCookie, setCookie, deleteCookie, hasCookie, validateContextCookies } from './index';
+import type { OptionsType, AppRouterCookies } from './types';
+import { NextRequest, NextResponse } from 'next/server';
+
+// Mock implementations
+const mockAppRouterCookies: AppRouterCookies = {
+  getAll: jest.fn().mockReturnValue([
+    { name: 'test', value: 'value' },
+    { name: 'jsonCookie', value: JSON.stringify({ foo: 'bar' }) }
+  ]),
+  set: jest.fn(),
+  get: jest.fn(),
+  has: jest.fn(),
+  delete: jest.fn(),
+  clear: jest.fn(),
+  [Symbol.iterator]: jest.fn(),
+  size: 2,
+};
+
+const mockOptions = {
+  req: {
+    cookies: mockAppRouterCookies,
+  } as unknown as NextRequest,
+  res: {
+    cookies: mockAppRouterCookies,
+  } as unknown as NextResponse,
+  cookies: jest.fn().mockResolvedValue(mockAppRouterCookies),
+} satisfies OptionsType;
+
+describe('Cookie Functions', () => {
+  beforeEach(() => {
```

Comment:
```suggestion
  afterEach(() => {
```
I think it would be better to clean everything after the test is done